### PR TITLE
Fix broken doc links, clean up irrelevant/old docs

### PR DESCRIPTION
**Pull request by dwhitena**
Fixes https://github.com/pachyderm/pachyderm/issues/1769.

Also:

- Updates the Getting Started overview doc, which was actually broken in multiple ways.
- Get rid of irrelevant and old docs including FAQ and troubleshooting under getting started (Note that troubleshooting in getting started was not up-to-date.  If we want to have this, we should redo it. It was more confusing than anything).
- Fixes formatting issues in the pachctl docs.
- Adds `copy-file` to the pachctl docs.

### CODE REVIEW

File: doc/deployment/google_cloud_platform.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ $ gcloud compute disks list
 $ brew tap pachyderm/tap && brew install pachyderm/tap/pachctl@1.6
 
 # For Linux (64 bit):
-$ curl -o /tmp/pachctl.deb -L https://github.com/pachyderm/pachyderm/releases/download/v1.6.0-RC1/pachctl_1.6.0-RC1_amd64.deb && sudo dpkg -i /tmp/pachctl.deb
+$ curl -o /tmp/pachctl.deb -L https://github.com/pachyderm/pachyderm/releases/download/v1.6.0-RC2/pachctl_1.6.0-RC2_amd64.deb && sudo dpkg -i /tmp/pachctl.deb
```
Comment (JoeyZwicker) [inline doc/deployment/google_cloud_platform.md]: we're on rc4? does this need to be manually updated for every release? I thought we had this managed automatically?

**Comment by JoeyZwicker** [comment]
What do you mean by Troubleshooting was more confusing than helpful? it's true that it needs updating, but I feel like the goal there was add all the common beginner questions? What would you want it to be instead -- I guess something more like stack overflow?

**Comment by JoeyZwicker** [comment]
LGTM. I'm not opposed to taking out troubleshooting, but I worry we'll see an uptick of very common user issues so we'll likely need something similar

**Comment by dwhitena** [comment]
@JoeyZwicker I agree that we should probably generate something new for beginner troubleshooting.  I was just meaning that if they are already struggling, and then they see stuff about grep and fruit stand, it might be worse than just removing them for now and revisiting.

**Pull request merged**